### PR TITLE
Add basic agent logging preamble

### PR DESCRIPTION
**Pull request by z1c0**
This follows the spec for a logging preambel in this [draft PR](https://github.com/elastic/apm/pull/713/).

### CODE REVIEW

File: src/Elastic.Apm/AgentComponents.cs [hunk 1]
```diff
@@ -182,5 +187,24 @@ public void Dispose()
...
 		}
+
+		private static void PrintAgentLogPreambel(IApmLogger logger)
+		{
+			if (logger?.Info() != null)
+			{
+				var info = logger.Info().Value;
+				info.Log("********************************************************************************");
+				info.Log(
+					$"Elastic APM .NET Agent, version: {Assembly.GetExecutingAssembly().GetName().Version}, build date: {File.GetCreationTime(Assembly.GetExecutingAssembly().Location).ToUniversalTime()} UTC");
```
Comment (z1c0) [inline src/Elastic.Apm/AgentComponents.cs:198]: The solution for the "build date" is not yet ideal, but a good first approximation.
A better mid- to long-term solution could be to introduce a custom assembly attribute that gets actually set a build time.

File: src/Elastic.Apm/AgentComponents.cs [hunk 2]
```diff
@@ -182,5 +187,24 @@ public void Dispose()
 
 			CentralConfigurationFetcher?.Dispose();
 		}
+
+		private static void PrintAgentLogPreambel(IApmLogger logger)
```
Comment (gregkalapos) [inline src/Elastic.Apm/AgentComponents.cs:191]: nit: small typo I think.

```suggestion
		private static void PrintAgentLogPreamble(IApmLogger logger)
```

File: src/Elastic.Apm/AgentComponents.cs (hunk 1 above)
Comment (gregkalapos) [reply src/Elastic.Apm/AgentComponents.cs:198]: 👍  I also think this is a good first start. Just one thing maybe: I think it'd be better to print "agent file creation time" (or something along those lines) to make sure there is no misunderstanding by this log line.

**Review by gregkalapos — COMMENTED** [review]
I see some of the properties can throw exceptions which we access.  According to the doc, all are `InvalidOperationException`s. 

Here are the ones:
- `Process.GetCurrentProcess().Id` (the `Id` property)
- `Process.GetCurrentProcess().ProcessName` (again the `ProcessName` property)
- `Environment.MachineName`

Super unlikely cases, but I'd feel better by having a try-catch on those.

File: src/Elastic.Apm/AgentComponents.cs (hunk 1 above)
Comment (z1c0) [reply src/Elastic.Apm/AgentComponents.cs:198]: Agreed, then this will be unambiguous 👍

**Comment by z1c0** [comment]
> I see some of the properties can throw exceptions which we access. According to the doc, all are `InvalidOperationException`s.
> 
> Here are the ones:
> 
> * `Process.GetCurrentProcess().Id` (the `Id` property)
> * `Process.GetCurrentProcess().ProcessName` (again the `ProcessName` property)
> * `Environment.MachineName`
> 
> Super unlikely cases, but I'd feel better by having a try-catch on those.

Good catch (pun intended)!
I'll wrap the whole preamble block in a `try/catch` statement - that way, it's less likely that we (I 😉) miss exceptions that might be thrown by log statements added in the future.

**Review by gregkalapos — APPROVED** [review]
Thanks!

**Pull request merged**